### PR TITLE
CIT-62: Review and submission page

### PR DESCRIPTION
**Pull request by zacpez**
* Added business contact card: component, story, tests

### CODE REVIEW

**Comment by zacpez** [comment]
Do not squash merge, if possible

File: cit3.0-web/src/components/OpportunityView/Resource.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    // Individual data points
+    if (
+      resource[1].subtitle &&
+      typeof toDisplay[resource[1].title][resource[1].subtitle] === "undefined"
+    ) {
+      toDisplay[resource[1].title][resource[1].subtitle] = {};
+    }
+
+    // Different markup for datapoint types
+    /* eslint-disable no-fallthrough */
```
Comment (BrendanBeachBC) [inline cit3.0-web/src/components/OpportunityView/Resource.js:34]: Are we able to remove this disable line? I looked it up and this code appears to be following the rules correctly. Was it still causing errors for some reason?

File: cit3.0-web/src/components/Page/AddOpportunity/AddOpportunity.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  const coords = useSelector((state) => state.opportunity.coords);
+  const nearbyResources = useSelector(
+    (state) => state.opportunity.nearbyResources
+  );
+
+  const history = useHistory();
+  const title1 = "Add an Opportunity";
+  const title2 = "Confirm Property";
+  const text1 =
+    "Please drop a min on the map or enter the address of the property you want to list as an opportunity for investors.";
```
Comment (BrendanBeachBC) [inline cit3.0-web/src/components/Page/AddOpportunity/AddOpportunity.js:32]: I think this is a typo
```suggestion
    "Please drop a pin on the map or enter the address of the property you want to list as an opportunity for investors.";
```

File: cit3.0-web/src/components/Page/ReviewOpportunity/ReviewOpportunity.stories.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+/* eslint-disable */
```
Comment (BrendanBeachBC) [inline cit3.0-web/src/components/Page/ReviewOpportunity/ReviewOpportunity.stories.js:1]: Same as earlier, why disable for this file?

File: cit3.0-web/src/index.css [hunk 4]
```diff
@@ -1,6 +1,9 @@
 html, body {
   margin: 0;
   padding: 0;
+  /* height: 100vh; */
```
Comment (BrendanBeachBC) [inline cit3.0-web/src/index.css:4]:
```suggestion

```

Comment (BrendanBeachBC) [inline cit3.0-web/src/index.css:4]:
```suggestion

```

File: cit3.0-web/src/index.css [hunk 5]
```diff
@@ -13,11 +16,17 @@ body {
     sans-serif;
   -webkit-font-smoothing: antialiased;
   -moz-osx-font-smoothing: grayscale;
-  height: 100vh;
-  width: 100vw;
+  /* height: 100%; */
+  /* width: 100vw; */
```
Comment (BrendanBeachBC) [inline cit3.0-web/src/index.css:20]:
```suggestion

```

File: cit3.0-web/src/setupTests.js [hunk 6]
```diff
@@ -2,19 +2,21 @@
 // allows you to do things like:
 // expect(element).toHaveTextContent(/react/i)
 // learn more: https://github.com/testing-library/jest-dom
+/* eslint-disable */
```
Comment (BrendanBeachBC) [inline cit3.0-web/src/setupTests.js:5]: Why is eslint disabled for this file?

File: cit3.0-web/src/setupTests.js [hunk 7]
```diff
@@ -2,19 +2,21 @@
...
+//   if (
+//     namespaceURI === "http://www.w3.org/2000/svg" &&
+//     qualifiedName === "svg"
+//   ) {
+//     const element = createElementNSOrig.apply(this, arguments);
+//     element.createSVGRect = () => {};
+//     return element;
+//   }
+//   return createElementNSOrig.apply(this, arguments);
+// };
```
Comment (BrendanBeachBC) [inline cit3.0-web/src/setupTests.js:22]: Do we still need this code? 
```suggestion

```

**Review by BrendanBeachBC — COMMENTED** [review]
Looks pretty good to me! I left some comments for some minor things to change or discuss, but looks good to merge in after we have those conversations!

File: cit3.0-web/src/components/ButtonRow/ButtonRow.stories.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+  <ButtonRow noContinue={false} onClick={() => action("clicked")} />
+);
+
+export const ButtonRowWithPreviousPageLink = () => (
+  <ButtonRow
+    noContinue={false}
+    previousPage="/addOpportunity/siteDetails"
+    onClick={() => action("clicked")}
+  />
+);
```
[the remark this answers is not in the document]
Comment (zacpez) [reply cit3.0-web/src/components/ButtonRow/ButtonRow.stories.js:27]: Changed, you will see the change in the next commit

File: cit3.0-web/src/components/Headers/NavigationHeader/NavigationHeader.stories.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+const navItems = [
+  "Location",
+  "Property Details",
+  "Additional Info",
+  "Review & Submit",
+];
+
+export const NavigationHeaderDisplay = () => (
+  <NavigationHeader navItems={navItems} />
```
[the remark this answers is not in the document]
Comment (zacpez) [reply cit3.0-web/src/components/Headers/NavigationHeader/NavigationHeader.stories.js:17]: Fixed, missing required param currentStep on the story.

File: cit3.0-web/src/components/Headers/NavigationHeaderItem/NavigationHeaderItem.stories.js [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+import NavigationHeaderItem from "./NavigationHeaderItem";
+
+export default {
+  title: "Navigation Header Item",
+  component: NavigationHeaderItem,
+};
+
+export const NavigationHeaderItemDisplay = () => (
+  <div className="py-4" style={{ backgroundColor: "#E0E0E0" }}>
+    <NavigationHeaderItem label="Location" step="1" />
```
[the remark this answers is not in the document]
Comment (zacpez) [reply cit3.0-web/src/components/Headers/NavigationHeaderItem/NavigationHeaderItem.stories.js:11]: Fixed, missing required params `step` as number, and missing `currentStep` on the story.

File: cit3.0-web/src/components/LocationsPanel/LocationsPanel.stories.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
+/* eslint-disable */
```
[the remark this answers is not in the document]
Comment (zacpez) [reply cit3.0-web/src/components/LocationsPanel/LocationsPanel.stories.js:1]: Not needed anymore.

File: cit3.0-web/src/components/LocationsPanel/LocationsPanel.js [hunk 12]
```diff
@@ -0,0 +1,131 @@
+import React from "react";
+import PropTypes from "prop-types";
+import NumberFormat from "react-number-format";
+import { Container, Row, Col } from "react-bootstrap";
+import { v4 } from "uuid";
+import styles from "./LocationsPanel.module.css";
+
+/** eslint-disable react/prop-types */
+const LocationsPanel = ({
```
[the remark this answers is not in the document]
Comment (zacpez) [reply cit3.0-web/src/components/LocationsPanel/LocationsPanel.js:9]: Not needed anymore.

File: cit3.0-web/src/index.css (hunk 4 above)
Comment (zacpez) [reply cit3.0-web/src/index.css:4]: Removed in next commit.

File: cit3.0-web/src/index.css (hunk 5 above)
Comment (zacpez) [reply cit3.0-web/src/index.css:20]: Removed in next commit.

File: cit3.0-web/src/setupTests.js (hunk 7 above)
Comment (zacpez) [reply cit3.0-web/src/setupTests.js:22]: Maybe, we'd need to consult with Sarah. There was good purpose.

File: cit3.0-web/src/components/LocationsPanel/LocationsPanel.stories.js [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+    lng: 48.785009
+};
+
+const munis = [
+    {name: "Duncan", link: "www.example.com", distance: 1.23},
+    {name: "Duncan", link: "www.example.com", distance: 4.56},
+    {name: "Duncan", link: "www.example.com", distance: 7.89},
+];
+
+storiesOf('LocationsPanel', module).add('default', () => <LocationsPanel address="3045 Gibbins Road, Duncan, BC" coords={coords} municipalities={munis} />);
```
[the remark this answers is not in the document]
Comment (zacpez) [reply cit3.0-web/src/components/LocationsPanel/LocationsPanel.stories.js]: Generated story, will have to see if we can update the template.

File: cit3.0-web/src/components/Map/Map.js [hunk 14]
```diff
@@ -5,14 +5,16 @@ import {
...
   MapConsumer,
 } from "react-leaflet";
+import L from "leaflet";
+
 import "./map.css";
 import ChangeView from "../ChangeView/ChangeView";
 import AddLocationMarker from "../AddMarker/AddMarker";
 import ResourceMarker from "../AddMarker/ResourceMarker";
 
+/* eslint-disable no-unused-vars */
```
[the remark this answers is not in the document]
Comment (zacpez) [reply cit3.0-web/src/components/Map/Map.js:17]: Removed one comment, need this to enable a following ticket.

File: cit3.0-web/src/components/Map/Map.js [hunk 15]
```diff
@@ -52,7 +70,7 @@ export default function Map({
           setNearbyResources={setNearbyResources}
           changeView={changeView}
         />
-        <Polygon pathOptions={{ color: "purple" }} positions={multiPolygon} />
+        {/* <Polygon pathOptions={{ color: "purple" }} positions={multiPolygon} /> */}
```
[the remark this answers is not in the document]
Comment (zacpez) [reply cit3.0-web/src/components/Map/Map.js:73]: Removed one comment, need this to enable a following ticket.

File: cit3.0-web/src/components/Map/Map.js [hunk 16]
```diff
@@ -32,6 +34,22 @@ export default function Map({
   let zoomLevel;
 
   const changeView = (centerCoords) => centerCoords;
+  const [geoData, setGeoData] = useState({});
+  const [bounds, setBounds] = useState("");
+  // const [url, setUrl] = useState("");
```
[the remark this answers is not in the document]
Comment (zacpez) [reply cit3.0-web/src/components/Map/Map.js:39]: Removed the unused state

File: cit3.0-web/src/components/MapContainer/MapContainer.test.js [hunk 17]
```diff
@@ -69,66 +72,36 @@ describe("MapContainer", () => {
...
-          setNearbyResources={setNearbyResources}
-        />
-      );
-      const searchButton = container.querySelector("button");
-      const addressInput = queryByPlaceholderText("Address");
-      fireEvent.change(addressInput, changeEvent(address));
-      fireEvent.click(searchButton);
-      expect(axios.get).toHaveBeenCalledTimes(1);
+describe("api call", () => {
+  xit("returns data from axios call", async () => {
```
[the remark this answers is not in the document]
Comment (zacpez) [reply cit3.0-web/src/components/MapContainer/MapContainer.test.js:84]: Removed, as the use case is no longer valid

File: cit3.0-web/src/components/OpportunityView/Resource.js (hunk 1 above)
Comment (zacpez) [reply cit3.0-web/src/components/OpportunityView/Resource.js:34]: Removed, as a previous solution needed it

File: cit3.0-web/src/components/Page/AddOpportunity/AddOpportunity.js (hunk 2 above)
Comment (zacpez) [reply cit3.0-web/src/components/Page/AddOpportunity/AddOpportunity.js:32]: Fixed the typo

File: cit3.0-web/src/components/Page/ReviewOpportunity/ReviewOpportunity.stories.js (hunk 3 above)
Comment (zacpez) [reply cit3.0-web/src/components/Page/ReviewOpportunity/ReviewOpportunity.stories.js:1]: Removed. Was used before Provider wrapper was added.

File: cit3.0-web/src/index.css (hunk 4 above)
Comment (zacpez) [reply cit3.0-web/src/index.css]: Removed.

File: cit3.0-web/src/setupTests.js (hunk 6 above)
Comment (zacpez) [reply cit3.0-web/src/setupTests.js]: Removed.

**Review by BrendanBeachBC — APPROVED** [review]

**Pull request merged**